openwillis/measures/video/eye_blink.py

```diff
@@ -0,0 +1,345 @@
+    left_ips : array
+        Array containing the frame number of the start of each blink
+    right_ips : array
+        Array containing the frame number of the end of each blink
+
+    ---------------------------------------------------------------------------------------------------
+    """
+    troughs, properties = find_peaks(-framewise['EAR'], prominence=prominence, width=width)
+
+    left_ips = properties["left_ips"]
+    right_ips = properties["right_ips"]
+
+    # round to nearest integer and add 1 to match frame number
+    left_ips = np.round(left_ips).astype(int) + 1
+    right_ips = np.round(right_ips).astype(int) + 1
+    troughs = np.round(troughs).astype(int) + 1
+
+    return troughs, left_ips, right_ips
+
+
+def convert_frame_to_time(troughs, left_ips, right_ips, fps, config):
+    """
+    ---------------------------------------------------------------------------------------------------
+
+    This function converts the frame number to time.
+
+    Parameters:
+    ............
+    troughs : array
+        Array containing the frame number of each blink
+    left_ips : array
+        Array containing the frame number of the start of each blink
+    right_ips : array
+        Array containing the frame number of the end of each blink
+    fps : float
+        The fps of the video
+    config : dict
+        The configuration dictionary
+
+    Returns:
+    ............
+    blinks : pd.DataFrame
+        Contains for each blink the frame number and the time (in seconds)
+         start of the blink and end of the blink
+
+    ---------------------------------------------------------------------------------------------------
+    """
+    troughs_time = troughs/fps
+    left_ips_time = left_ips/fps
+    right_ips_time = right_ips/fps
+
+    blinks = pd.DataFrame(
+        {
+            config['peak_frame']: troughs, config['start_frame']: left_ips, config['end_frame']: right_ips,
+            config['peak_time']: troughs_time, config['start_time']: left_ips_time, config['end_time']: right_ips_time
+        }
+    )
+    return blinks
+
+
+def calculate_framewise(vs, face_mesh, config):
+    """
+    ---------------------------------------------------------------------------------------------------
+
+    This function calculates the eye aspect ratio (EAR) of each frame.
+
+    Parameters:
+    ............
+    vs : object
+        The video capture object
+    face_mesh : object
+        The MediaPipe Face Mesh model
+    config : dict
+        The configuration dictionary
+
+    Returns:
+    ............
+    framewise : pd.DataFrame
+        Contains the frame number and the eye aspect ratio (EAR) of each frame
+    frame_n : int
+        The total number of frames
+
+    ---------------------------------------------------------------------------------------------------
+    """
+
+    framewise = []
+    frame_n = 0
+
+    while True:
+        ret, frame = vs.read()
```

Comment:
add try catch for this while loop

Reply:
[add logic to handle error in while loop of ear calculation](https://github.com/bklynhlth/openwillis/pull/21/commits/eed9c735bb28334205f1d901654a5fd03e0586fc)